.github/actions/pack-build/action.yml

```diff
@@ -0,0 +1,61 @@
+name: 'Pack Build'
+description: 'Pack build images'
+
+inputs:
+  tag:
+    description: 'location to write image'
+    required: true
+  bp_go_targets:
+    description: 'value of BP_GO_TARGETS env'
+  builder:
+    description: 'builder image'
+    required: true
+    default: 'paketobuildpacks/builder-jammy-tiny'
+  pack_version:
+    description: 'version of pack to use'
+    required: true
+  additional_pack_args:
+    description: 'additional args for pack'
+
+outputs:
+  image:
+    description: "Built image"
+    value: ${{ steps.build.outputs.image }}
+  digest:
+    description: "Built image digest"
+    value: ${{ steps.build.outputs.digest }}
+
+runs:
+  using: "composite"
+  steps:
+  - name: setup-pack
+    uses: buildpacks/github-actions/setup-pack@v5.0.0
+    with:
+      pack-version: ${{ inputs.pack_version }}
+  - name: build
+    id: build
+    shell: bash
+    run: |
+      mkdir report
+      
+      export PATH="$PATH:$(pwd)"
+      pack build ${{ inputs.tag }} \
+        --builder ${{ inputs.builder }} \
+        --env BP_GO_TARGETS="${{ inputs.bp_go_targets }}" \
+        --report-output-dir . \
+        --cache-image ${{ inputs.tag }}-cache \
+        --publish ${{ inputs.additional_pack_args }}
+      
+      mkdir images
+      digest=$(go run .github/actions/pack-build/report.go -path ./report.toml)
```

Comment:
Turns out `yq` can handle toml now: `yq '.image.digest' report.toml`. (I don't have an opinion which way we do it, but I just found it interesting there's finally a stream processor for toml)

Reply:
Might be worth looking into